arithmetization/src/main/java/net/consensys/linea/zktracer/module/mmu/MmuOperation.java

```diff
@@ -159,24 +161,28 @@ public void fillLimb() {
       final boolean exoIsTarget = mmuData.exoLimbIsTarget();
       checkArgument(exoIsSource == !exoIsTarget, "ExoLimb is either the source or the target");
 
-      for (MmuToMmioInstruction mmioInst : this.mmuData.mmuToMmioInstructions()) {
-        final int offset =
-            (int)
-                (exoIsSource
-                    ? mmioInst.sourceLimbOffset() * LLARGE + mmioInst.sourceByteOffset()
-                    : mmioInst.targetLimbOffset() * LLARGE + mmioInst.targetByteOffset());
-        final int sizeToExtract = mmioInst.size() == 0 ? LLARGE : mmioInst.size();
-        final Bytes16 exoLimb = readBytes(mmuData.exoBytes(), offset, sizeToExtract);
-        mmioInst.limb(exoLimb);
+      for (MmuToMmioInstruction mmioInst : mmuData.mmuToMmioInstructions()) {
```

Comment:
this is a fix

Reply:
⭐

Reply:
if the exoBytes is the source, We were extracting the (exo)limb from the SBO which is totally wrong. The exo limb is 16 bytes starting at 16 * SLO? And from this exo limb, we extract from SBO to extract the data. But We weren't reading the right exo(limb) as we were taking 16 bytes from 16*SLO +  SBO for the exo bytes